components/mdns/mdns_networking.c

```diff
@@ -127,7 +127,12 @@ static void _udp_recv(void *arg, struct udp_pcb *upcb, struct pbuf *pb, const ip
         packet->tcpip_if = MDNS_IF_MAX;
         packet->pb = this_pb;
         packet->src_port = rport;
-        memcpy(&packet->src, raddr, sizeof(ip_addr_t));
+        packet->src.type = raddr->type;
+        if (raddr->type == IPADDR_TYPE_V4) {
```

Comment:
Is this condition correct? seems to me we would like to copy the IPv4 address if the type is `IPADDR_TYPE_V4`